app/__init__.py

```diff
@@ -289,49 +282,47 @@ def make_session_permanent():
 
 
 def load_service_before_request():
+    g.current_service = None
+
     if "/static/" in request.url:
-        _request_ctx_stack.top.service = None
         return
-    if _request_ctx_stack.top is not None:
-        _request_ctx_stack.top.service = None
 
-        if request.view_args:
-            service_id = request.view_args.get("service_id", session.get("service_id"))
-        else:
-            service_id = session.get("service_id")
+    if request.view_args:
+        service_id = request.view_args.get("service_id", session.get("service_id"))
+    else:
+        service_id = session.get("service_id")
 
-        if service_id:
+    if service_id:
+        try:
+            g.current_service = Service(service_api_client.get_service(service_id)["data"])
```

Comment:
I know you’ve copied this from the existing code, but would be nicer as this (and matches `Organisation.from_id(org_id)` further down): 
```suggestion
            g.current_service = Service.from_id(service_id)
```